sdk/lib/opentelemetry/sdk/trace/export/batch_span_processor.rb

```diff
@@ -174,6 +175,10 @@ def lock
               yield
             end
           end
+
+          def forking_dispatcher?
+            %i[passenger puma rainbows unicorn].include?(@dispatcher)
```

Comment:
This is brittle, and this knowledge should not be in the SDK. If the functionality is necessary (and I'm not really convinced it is), we should find a way to configure this in the instrumentation for each of these runtimes rather than hardcoding a list in the SDK.

Reply:
This is the exact way it is implemented in other telemetry libs. It's "brittle" only in the sense that if someone creates a new forking ruby server we'd need to add it to the list; it's not brittle in the sense that it will spontaneously break for these specific widely-used servers. As far as I'm aware there is no way to pre-emptively know that a process is going to fork, unless we [violate the laws of causality](https://www.youtube.com/watch?v=AZGcmvrTX9M)

Reply:
> It's "brittle" only in the sense that if someone releases a new forking ruby server we'd need to add it to the list

This is exactly the problem. For example, a quick search turns up https://github.com/bdurand/sidekiq-process_manager for Sidekiq. We have no idea what pre-forking runtimes exist as private repos. It should not be necessary to fork OpenTelemetry Ruby or to monkey-patch the SDK in order to support additional pre-forking runtimes.

Reply:
"sidekiq-process_manager" is an unofficial hack gem which is not part of the official Sidekiq release. For these non-standard use cases, I suppose we could add an env var such as OTEL_FORKING_DISPATCHER to force a behavior one way or the other.

In such cases there is no hard issue with spawning the thread pre-fork, aside from wasting a bit of CPU and possibly seeing a annoying warning log.

Reply:
Is it fair to say the approach should be:
1. Add an env var (`OTEL_FORKING_DISPATCHER`) to explicitly control behavior
2. Fallback to detecting the dispatcher runtime as this PR does, to "do the right thing" automatically in 95% of cases